### PR TITLE
Make "use $crate" a hard error

### LINKED ISSUE #37390: Tracking issue for `use $crate;` compatibility warning
The warning was introduced in #37361 to avoid breakage from #37213.

### PR DESCRIPTION
**Pull request by mental32**
Closes #37390

### CODE REVIEW

File: src/test/ui/dollar-crate/dollar-crate-is-keyword.stderr [hunk 1]
```diff
@@ -8,35 +8,30 @@ LL | m!();
...
    |         ^^^^^^^^^^^^^^^^^^^^^
 ...
 LL | m!();
    | ----- in this macro invocation
-   |
-   = note: `use $crate;` was erroneously allowed and will become a hard error in a future release
-
-error: aborting due to 2 previous errors
 
+error: aborting due to 4 previous errors
```
Comment (JohnTitor) [inline src/test/ui/dollar-crate/dollar-crate-is-keyword.stderr]: Needs a blank line
```suggestion
error: aborting due to 4 previous errors

```

File: src/test/ui/imports/import-crate-var.stderr [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
+  --> $DIR/import-crate-var.rs:6:5
    |
 LL |     m!();
    |     ^^^^^
    |
-   = note: `use $crate;` was erroneously allowed and will become a hard error in a future release
-   = note: this warning originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
+   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
 
+error: aborting due to previous error
```
Comment (JohnTitor) [inline src/test/ui/imports/import-crate-var.stderr]: Same here

```suggestion
error: aborting due to previous error

```

**Comment by mental32** [comment]
@JohnTitor Thanks!

First time contributor here and the UI tests are difficult to get used to, I'm unable to build Rustc so I cant test changes locally, so far its been hit and miss :(

**Comment by JohnTitor** [comment]
Uhm, applying suggestions on GitHub ignores blank lines, so could you insert them on your local and push it? Anyway, thanks for considering the contributions.

**Comment by JohnTitor** [comment]
Also, it's good to squash commits into one.

**Comment by Dylan-DPC** [comment]
yaay finally :P

@bors r+ rollup

**Pull request merged**